cpu/stm32_common/periph/pm.c

```diff
@@ -39,31 +42,56 @@
  *
  * Available values can be found in reference manual, PWR section, register CR.
  */
+#if defined(CPU_FAM_STM32F0)
+#define PM_STOP_CONFIG (PWR_CR_LPDS)
+#else
 #define PM_STOP_CONFIG (PWR_CR_LPDS | PWR_CR_FPDS)
 #endif
+#endif
+
+#ifndef PM_EWUP_CONFIG
+/**
+ * @brief   Define EWUP config flags
+ *
+ * Available values can be found in reference manual, PWR section, register CSR.
+ */
+#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || \
+    defined(CPU_FAM_STM32F4)
+#define PM_EWUP_CONFIG  (PWR_CSR_EWUP)
+#elif defined(PWR_CSR_EWUP8)
+#define PM_EWUP_CONFIG  (PWR_CSR_EWUP8 | PWR_CSR_EWUP7 | PWR_CSR_EWUP6 | \
```

Comment:
Maybe readability can be improved by using a static inline?

```
static const unsigned _pwr_ewup_config(void)
{
unsigned tmp = 0;
#if defined(PWR_CSR_EWUP8)
tmp |= PWR_CSR_EWUP8;
#endif
#if defined(PWR_CSR_EWUP7)
tmp |= PWR_CSR_EWUP7;
#endif
...
#if defined(PWR_CSR_EWUP1)
tmp |= PWR_CSR_EWUP1;
#endif

return tmp;
}
```

... then use ```PWR->CSR = _pwr_ewup_config()``` below.

This *should* optimize to a single value even when assigning a ```volatile``` register.

Reply:
I'll give a try

Reply:
Done